sagemaker-debugger/tensorflow_action_on_rule/detect_stalled_training_job_and_stop.ipynb

```diff
@@ -4,30 +4,22 @@
    "cell_type": "markdown",
```

Comment:
<p>For DLC you don't need to provide Collections to save.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/1592/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Reply:
<p>I don't think the StalledTrainingRule has a default tensor collection to watch, so I specified one as an example. It has to have a tensor collection to watch (based on the <a href="https://github.com/awslabs/sagemaker-debugger-rulesconfig/blob/master/smdebug_rulesconfig/rule_config_jsons/ruleConfigs.json" target="_blank">ruleConfigs.json</a> file). For DLC, based on my understanding, it is required to specify a collection either throught the <code>Rule.sagemaker</code>'s collections_to_save or the <code>DebugHookConfig()</code> API, unless the rule is pre-configured as provided in the <a href="https://github.com/awslabs/sagemaker-debugger-rulesconfig/blob/master/smdebug_rulesconfig/rule_config_jsons/ruleConfigs.json" target="_blank">ruleConfigs.json</a> file.</p>

Reply:
> I don't think the StalledTrainingRule has a default tensor collection to watch, so I specified one as an example. Each rule has to have a tensor collection to watch (based on the [ruleConfigs.json](https://github.com/awslabs/sagemaker-debugger-rulesconfig/blob/master/smdebug_rulesconfig/rule_config_jsons/ruleConfigs.json) file, some have default collections, some don't). For DLC, based on my understanding, it is required to specify a collection either through the `Rule.sagemaker`'s `collections_to_save` or the `DebugHookConfig()` API, unless the rule is pre-configured as provided in the [ruleConfigs.json](https://github.com/awslabs/sagemaker-debugger-rulesconfig/blob/master/smdebug_rulesconfig/rule_config_jsons/ruleConfigs.json) file.

Stalled training rule checks if there is 'any' tensor emitted. By default in DLC, we have loss emitted every 500 steps, so it should just work. 
When you don't have DLC, you would need to emit tensor on your own.

Reply:
<p>For example, vanishing_gradient() or loss_not_decreasing() rules are having <code>gradients</code>  and <code>losses</code>  tensor collections by default in their .json config file, so we don't have to specify any tensor collection. But the ones without the default collections, I think it needs to be specified</p>

Reply:
Ok, can you try running the notebook after removing collection to save?

Reply:
https://github.com/aws/amazon-sagemaker-examples/pull/1592#discussion_r499962051
If this is true, I would say saving `losses` tensors using DebugHookConfig() should conflict. But that never happened.

For example, if I include `vanishing_gradient()` rule and add DebuggerHookConfig(collection_configs="gradients"), then it returns a confliction error, saying that the duplicated collections names are used.

Reply:
Does Debugger collect tensors with regex ".*loss" by default, or the "losses" collection by default?

Reply:
> Ok, can you try running the notebook after removing collection to save?
Yes, I can run it without the collection to save, but the training job has the 10 minute sleep time, so it will not change the result tho?

Reply:
2 minute threshold would fire the rule, if there is 10 minute sleep in training.

Reply:
Yep that's what I meant. The training job ran and slept well to stop after 2 mins, and it will be the same for with and without the collection_to_save option. Then do you think we need to remove it to simplify the rule config?

Reply:
yes, if users are using DLC they don't  need to add collections to save for this rule. Though we can make it clear, why? Saying that smdebug default saves losses every 500 steps, which will act as heartbeat for training job.

Reply:
Incorporated

Reply:
I guess  it did worked when you removed collection configuration, is that right?

Reply:
Yes, it did.